### PR TITLE
improve(polardb): feature polardb support connection

### PR DESCRIPTION
**Pull request by zhangyuangang**
image
image

### CODE REVIEW

File: alicloud/data_source_alicloud_polardb_endpoints.go [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+			"db_cluster_id": {
+				Type:     schema.TypeString,
+				ForceNew: true,
+				Required: true,
+			},
+			"db_endpoint_id": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+
```
Comment (Lexsss) [inline alicloud/data_source_alicloud_polardb_endpoints.go]: ids

File: alicloud/resource_alicloud_polardb_endpoint.go [hunk 2]
```diff
@@ -0,0 +1,247 @@
...
+func resourceAlicloudPolarDBEndpointDelete(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*connectivity.AliyunClient)
+	polarDBService := PolarDBService{client}
+
+	submatch := polardbEndpointIdWithSuffixRegexp.FindStringSubmatch(d.Id())
+	if len(submatch) > 1 {
+		d.SetId(submatch[1])
+	}
+
+	split := strings.Split(d.Id(), COLON_SEPARATED)
```
Comment (Lexsss) [inline alicloud/resource_alicloud_polardb_endpoint.go]: parse resource id

File: website/docs/r/polardb_endpoint.html.markdown [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+* `db_cluster_id` - (Required, ForceNew) The Id of cluster that can run database.
+* `db_endpoint_id` - (Required, ForceNew) The Id of endpoint that can run database.
+* `connection_prefix` - (Optional) Prefix of an Internet connection string. It must be checked for uniqueness. It may consist of lowercase letters, numbers, and underlines, and must start with a letter and have no more than 30 characters. Default to <db_endpoint_id> + 'tf'.
+* `net_type` - (Optional,ForceNew) Internet connection net type. Valid value: `Public`,`Private`. Default to `Public`.
+
+## Attributes Reference
+
+The following attributes are exported:
+
+* `id` - The current instance connection resource ID. Composed of instance ID and connection string with format `<db_cluster_id>:<db_endpoint_id>:<net_type>:<connection_prefix>`.
```
Comment (Lexsss) [inline website/docs/r/polardb_endpoint.html.markdown]: sync with create func

File: alicloud/resource_alicloud_polardb_endpoint_address.go [hunk 4]
```diff
@@ -0,0 +1,249 @@
...
+
+func resourceAlicloudPolarDBEndpointAddressCreate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*connectivity.AliyunClient)
+	polarDBService := PolarDBService{client}
+	clusterId := d.Get("db_cluster_id").(string)
+	dbEndpointId := d.Get("db_endpoint_id").(string)
+	netType := d.Get("net_type").(string)
+	prefix := d.Get("connection_prefix").(string)
+	if prefix == "" {
+		prefix = fmt.Sprintf("%stf", dbEndpointId)
```
Comment (xiaozhu36) [inline alicloud/resource_alicloud_polardb_endpoint_address.go]: Please remove the default value and use system default value.

File: alicloud/resource_alicloud_polardb_endpoint_address_test.go [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+				}),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheck(nil),
+				),
+			},
+			{
+				ResourceName:            resourceId,
+				ImportState:             true,
+				ImportStateVerify:       true,
+				ImportStateVerifyIgnore: []string{"connection_prefix"},
```
Comment (xiaozhu36) [inline alicloud/resource_alicloud_polardb_endpoint_address_test.go]: Please remove the line 56

Comment (zhangyuangang) [reply alicloud/resource_alicloud_polardb_endpoint_address_test.go]: > Please remove the line 56

The connection_prefix can not describe

File: alicloud/resource_alicloud_polardb_endpoint_address.go (hunk 4 above)
Comment (zhangyuangang) [reply alicloud/resource_alicloud_polardb_endpoint_address.go]: > Please remove the default value and use system default value.

How can I do ? There are examples ?

File: alicloud/data_source_alicloud_polardb_endpoints.go [hunk 6]
```diff
@@ -0,0 +1,158 @@
...
+									},
+									"port": {
+										Type:     schema.TypeString,
+										Computed: true,
+									},
+									"vpc_id": {
+										Type:     schema.TypeString,
+										Computed: true,
+									},
+									"v_switch_id": {
```
Comment (xiaozhu36) [inline alicloud/data_source_alicloud_polardb_endpoints.go]: Please use  `vswitch_id`

File: alicloud/data_source_alicloud_polardb_endpoints.go [hunk 7]
```diff
@@ -0,0 +1,158 @@
...
+			"address_items":      addrs,
+		}
+
+		s = append(s, mapping)
+	}
+	if err := d.Set("endpoints", s); err != nil {
+		return WrapError(err)
+	}
+
+	d.SetId(d.Get("db_cluster_id").(string))
```
Comment (xiaozhu36) [inline alicloud/data_source_alicloud_polardb_endpoints.go]: Please use all of endpoints id to generate a unique id

**Pull request merged**